enhancements/local-storage/automatic-detection.md

```diff
@@ -0,0 +1,460 @@
+The idea is to introduce two new features in LSO
+  - Auto discovery of local devices
+    - This will involve the intoduction of two new CRs `LocalVolumeDiscovery` and `LocalVolumeNodeDiscovery`.
+    - The purpose of this will be to expose local devices available in a node via the `LocalVolumeNodeDiscovery` CR.
+    - The device discovery will be done once via a discovery job. If an update happens to the `LocalVolumeNodeDiscovery` CR, the jobs gets retriggered
+  - Auto provisioning of localDevices
+    - This will involve introduction of a new CR called `AutoDetectVolume`.
+    - The pupose of this will be to auto discover and provision PVs on devices which match the inclusion filter present in the CR.
+    - This will involve a continous process of discovery of devices via the diskmaker daemons. Any discovered devices which matches the inclusion filter will be considered for provisioning of PVs.
+
+Once we have the detected devices the administrator can create the localPVs by explicitly selecting the disks. This can be done by the `localVolume` CR. The other option would be to create localPVs via the AutoDetectVolume CR and passing the inclusion filters in it.
+
+
+#### Workflow of LocalPV creation via LocalVolume CR
+
+- Discovery: The user can choose to run discovery if they want to understand what devices are available across the nodes in the cluster.
+    - ##### NOTE: This step is optional for users directly creating CRs. They might already know what CRs they need for step 2.
+- PV Creation: After the user has decided how to created PVs, they can choose to create PVs with either the LocalVolume CR or the AutoDetectVolume CR.
+
+
+## Design Details for `Auto discovery of local devices`
+
+API scheme for `LocalVolumeDiscovery` and `LocalVolumeNodeDiscovery` CR:
+
+```go
+type LocalVolumeDiscoveryList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	Items           []LocalVolumeDiscovery `json:"items"`
+}
+
+type LocalVolumeDiscovery struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	Spec              LocalVolumeDiscoverySpec   `json:"spec"`
+	Status            LocalVolumeDiscoveryStatus `json:"status,omitempty"`
+}
+
+type LocalVolumeDiscoverySpec struct {
+	// Nodes on which the automatic detection policies must run.
+	// +optional
+	NodeSelector *corev1.NodeSelector `json:"nodeSelector,omitempty"`
+	// If specified tolerations is the list of toleration that is passed to the
+	// LocalVolumeDiscovery Daemon
+	// +optional
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+}
+
+type LocalVolumeDiscoveryStatus struct {
+	Phase string `json:"phase,omitempty"`
+}
+
+type LocalVolumeNodeDiscoveryList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	Items           []LocalVolumeNodeDiscovery `json:"items"`
+}
+
+type LocalVolumeNodeDiscovery struct {
```

Comment:
I struggle with the name `LocalVolumeNodeDiscovery`. It's easy to confuse with `LocalVolumeDiscovery`. IMO, this object tracks *result* of the discovery per node, so should it be `LocalVolumeNodeDiscoveryResult`?

Reply:
done